WebAPI/Controllers/ExchangeController.cs

```diff
@@ -35,5 +35,18 @@ public async Task<IActionResult> ExchangeConfirm([FromBody]ExchangeRequestModel
             var res = await _blockChangesService.AddAndFindMatch(blockChangeRequest);
             return Ok(res);
         }
+        
+        [HttpPost("userWaitingExchanges")]
+        public async Task<IActionResult> GetUserWaitingExchanges([FromBody] string studentId)
+        {
+            bool isValidGUID = Guid.TryParse(studentId, out Guid guid);
+            if (!isValidGUID)
+            {
+                return ErrorResponse($"Student id: {studentId} is not valid GUID.");
+            }
+
+            var response = await _blockChangesService.FindWaitingStudentRequests(guid);
+            return Ok(response);
+        }
```

Comment:
everybody can ask for waiting exchange for everybody? Or do we have blocking somewhere else?
In my opinion, we need just GetUserWaitingExchanges without input parameters and you will return it for currently loged user..